src/flashbar/collapsible.scss

```diff
@@ -0,0 +1,229 @@
+/*
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ SPDX-License-Identifier: Apache-2.0
+*/
+@use '../internal/generated/custom-css-properties/index.scss' as custom-props;
+@use '../internal/hooks/focus-visible' as focus-visible;
+@use '../internal/styles/tokens' as awsui;
+@use '../internal/styles' as styles;
+@use '../internal/styles/typography' as typography;
+@use './collapsible.motion';
+
+.stack {
+  display: grid;
+  grid-template-columns: 1fr minmax(70px, auto) 1fr;
+}
+
+.stack > .expanded {
+  grid-column: 1 / 4;
+}
+
+@mixin stacked-flash-z-index {
+  // The first item should have the highest z index than the second item, and so forth...
+  z-index: calc(var(#{custom-props.$flashbarStackDepth}) - var(#{custom-props.$flashbarStackIndex}));
+}
+
+/*
+The collapsed stack is a dynamic grid that creates the necessary number of
+rows and columns based on the number of items passed to CSS via the
+`flashbarStackDepth` custom property. Assuming there are three flashbar items,
+the grid layout will be:
+
+[item 1 start] [10px] [10px] [fractional unit] [10px] [10px] [item 1 end]
+[10px] [item 2 start] [10px] [fractional unit] [10px] [item 2 end] [10px]
+[10px] [10px] [item 3 start] [fractional unit] [item 3 end] [10px] [10px]
+*/
+.stack > .collapsed {
+  display: grid;
+  grid-column: 1 / 4;
+  grid-template-columns:
+    repeat(var(#{custom-props.$flashbarStackDepth}), 10px)
+    1fr
+    repeat(var(#{custom-props.$flashbarStackDepth}), 10px);
+  row-gap: 8px;
+  z-index: 0;
+
+  > .item {
+    // Each item will have a unique custom property `flashbarStackIndex`
+    // Add +1 is because the array is indexed at zero but grid positions are indexed at 1.
+    grid-column-start: calc(var(#{custom-props.$flashbarStackIndex}) + 1);
+
+    // Add +1 to include the fractional unit in the column count
+    // Add +1 because the array indexed at zero but grid positions at indexed at 1.
+    grid-column-end: calc(
+      var(#{custom-props.$flashbarStackDepth}) * 2 + 1 - var(#{custom-props.$flashbarStackIndex}) + 1
+    );
+    grid-row-start: 1;
+    grid-row-end: calc(var(#{custom-props.$flashbarStackIndex}) + 2);
+
+    @include stacked-flash-z-index;
+  }
+
+  > .item:not(:first-of-type) {
+    align-self: end;
+    /* Give placeholder elements in the stacked state the height of a notification
+       with one single line of text.
+       This makes them look better during the collapse animation, in which they are already empty.
+     */
+    min-height: calc(
+      #{awsui.$font-body-m-line-height} + (
+          #{awsui.$space-scaled-xs} + #{awsui.$border-field-width} + #{awsui.$space-scaled-xxs}
+        ) * 2
+    );
+  }
+
+  &.visual-refresh {
+    > .item:not(:last-child) > .flash,
+    > .item.flash {
+      box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
```

Comment:
We usually want to turn these magic numbers into proper design tokens. Applies for other occurrences below as well.

Reply:
Added a design token for the actual whole shadow, as I saw there were other shadows defined as well. This also allowed to de-duplicate these two selector groups into one.